README.md

```diff
@@ -50,6 +50,28 @@ have dependencies on other algebras which must be implemented.
     - Two promises are equivalent when they yield equivalent values.
     - Two functions are equivalent if they yield equivalent outputs for equivalent inputs.
 
+## Type signature notation
+
+The type signature notation used in this document is described [here](https://sanctuary.js.org/#types). In summary:
+
+1. New types can be created via type constructors. Type constructors take one or more type arguments. For example, the type "array of strings" is written as `Array String`, and the type constructor `Array` can be defined as `Array :: a -> a`.
+2. Lowercase letters stand for type variables. Type variables can take any type unless they have been restricted by means of type constraints (see fat arrow below).
+3. `->` (arrow) expresses a function's explicit parameter and return value types. `String` -> `Array String` -> `Array String` is a function which takes a `String` and an `Array String` and returns an `Array String`.
+4. `~>` (squiggly arrow) expresses a function's implicit parameter type. When a function is a property of a type, it is called a method. All methods have an implicit parameter type - the type of which they are a property of.
+5. `=>` (fat arrow) expresses constraints on type variables. For example, `a -> b` is a function which takes an argument and returns a result. The argument can be any type. The result can be any type (including the same type as the argument). All this applies to the function described by `Functor f => f a -> f b` but we now require the type `f` to satisfy the Functor specification.
+
+i.e.
+
+```
+traverse :: Applicative f, Traversable t => t a ~> (TypeRep f, a -> f b) -> f (t b)
+'------'    '--------------------------'    '-'    '-------------------'    '-----'
+ '           '                               '      '                        ' - return type
+ '           '                               '      ' - argument type
+ '           '                               ' - method target type
+ '           ' - type constraint
+ '- method name
```

Comment:
@davidchambers Is this good:

```
map :: Functor f => f a ~> (a -> b) -> f b
'-'    '-------'    '-'    '------'    '-'
 '      '            '      '           '
 '- method name      ' - method target type
        '                   '           '
        ' - type constraint ' - argument type
                                        '
                                        ' - return type
```

Or should "return type" be level 1? Also, what about space between `type constraint ` and `argument type`? I kept it so small for consistency (all down arrows start from first "-" in the thing they underline).